### PR TITLE
add tolerations for gn-tools-pipeline

### PR DESCRIPTION
**Pull request by jeanmi151**
adding toleration to gn-tools-pipeline

### CODE REVIEW

File: gn-tools-pipelines/templates/gntoolspipelines-job.yaml [hunk 1]
```diff
@@ -15,11 +15,15 @@ spec:
         - name: {{ include "gn-tools-pipelines.fullname" . }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.config.tolerations }}
+          tolerations:
+            {{- .Values.config.tolerations | toYaml | nindent 8 }}
```
Comment (edevosc2c) [inline gn-tools-pipelines/templates/gntoolspipelines-job.yaml:20]: please put tolerations outside of the section "config". this is not a configuration about the application but about the Kubernetes definition for a pod.

see here for a good example:
- https://github.com/geonetwork/geonetwork-helm-charts/blob/main/datahub/values.yaml#L79
- https://github.com/geonetwork/geonetwork-helm-charts/blob/main/datahub/templates/datahub-deployment.yaml#L84-L86

**Review by edevosc2c — CHANGES_REQUESTED** [review]

File: gn-tools-pipelines/templates/gntoolspipelines-job.yaml [hunk 2]
```diff
@@ -15,11 +15,15 @@ spec:
         - name: {{ include "gn-tools-pipelines.fullname" . }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.tolerations }}
+          tolerations:
+            {{- .Values.tolerations | toYaml | nindent 8 }}
+          {{- end }}
```
Comment (edevosc2c) [inline gn-tools-pipelines/templates/gntoolspipelines-job.yaml:21]: can be simplified to:
```suggestion
          {{- with .Values.tolerations }}
          tolerations:
            {{- toYaml . | nindent 8 }}
          {{- end }}
```

**Review by edevosc2c — CHANGES_REQUESTED** [review]

File: gn-tools-pipelines/templates/gntoolspipelines-job.yaml [hunk 3]
```diff
@@ -15,11 +15,15 @@ spec:
         - name: {{ include "gn-tools-pipelines.fullname" . }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- with .Values.tolerations }}
+          tolerations:
```
Comment (edevosc2c) [inline gn-tools-pipelines/templates/gntoolspipelines-job.yaml:19]: I have just saw it but tolerations shouldn't be here. Tolerations applies to all containers in the pod so it should be in the definition of the pod not the container.

See https://blog.kubecost.com/blog/kubernetes-taints/

image

**Review by edevosc2c — CHANGES_REQUESTED** [review]

**Comment by jeanmi151** [comment]
@edevosc2c correct with all your suggestion, is it okay for you now ?

File: gn-tools-pipelines/templates/gntoolspipelines-job.yaml (hunk 3 above)
Comment (jeanmi151) [reply gn-tools-pipelines/templates/gntoolspipelines-job.yaml]: done

**Comment by jeanmi151** [comment]
> See above last request change.

done

**Review by edevosc2c — APPROVED** [review]

**Pull request merged**